scripts/specs/types/MobEntity.lua

```diff
@@ -20,6 +20,7 @@
 ---@field onMobFight? fun(mob: CBaseEntity, target: CBaseEntity)
 ---@field onCriticalHit? fun(mob: CBaseEntity, attacker: CBaseEntity?)
 ---@field onMobDeath? fun(mob: CBaseEntity, killer: CBaseEntity?, optParams: { isKiller: boolean, noKiller: boolean, isWeaponSkillKill: boolean, weaponskillUsed: xi.weaponskill, weaponskillDamage: integer })
+---@field onMobSpawnCheck? fun(mob: CBaseEntity)
```

Comment:
for non-optional return (which is probably what it should be)
```suggestion
---@field onMobSpawnCheck? fun(mob: CBaseEntity): integer
```

Reply:
Agreed it should require a return. Sorry for missing that on the first push but I'm afk for a few more hours at least

Reply:
yay, all good now